# SC1015 Project - Premier League Dataset

In [3]:
# Basic Libraries
import numpy as np
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt # we only need pyplot
sb.set() # set the default Seaborn style for graphics

In [4]:
resultsData = pd.read_csv('results.csv')
statsData = pd.read_csv('stats.csv')
statsData.head()

team  wins  losses  goals  total_yel_card  total_red_card  \
0  Manchester United  28.0     5.0   83.0            60.0             1.0   
1            Chelsea  24.0     3.0   64.0            62.0             4.0   
2          Liverpool  20.0    10.0   57.0            44.0             0.0   
3            Arsenal  19.0     8.0   63.0            59.0             3.0   
4  Tottenham Hotspur  17.0    12.0   57.0            48.0             3.0   

   total_scoring_att  ontarget_scoring_att  hit_woodwork  att_hd_goal  ...  \
0              698.0                 256.0          21.0         12.0  ...   
1              636.0                 216.0          14.0         16.0  ...   
2              668.0                 214.0          15.0          8.0  ...   
3              638.0                 226.0          19.0         10.0  ...   
4              520.0                 184.0           6.0          5.0  ...   

   total_cross  corner_taken  touches  big_chance_missed  clearance_off_line  \
0        918.0         258.0  25686.0                NaN                 1.0   
1        897.0         231.0  24010.0                NaN                 2.0   
2       1107.0         282.0  24150.0                NaN                 1.0   
3        873.0         278.0  25592.0                NaN                 1.0   
4        796.0         181.0  22200.0                NaN                 2.0   

   dispossessed  penalty_save  total_high_claim  punches     season  
0           NaN           2.0              37.0     25.0  2006-2007  
1           NaN           1.0              74.0     22.0  2006-2007  
2           NaN           0.0              51.0     27.0  2006-2007  
3           NaN           0.0              88.0     27.0  2006-2007  
4           NaN           0.0              51.0     24.0  2006-2007  

[5 rows x 42 columns]

## Total of 240 Rows and 42 Columns in stats.CSV file

### 2 Categorical Variables, 40 Numeric Variables

In [8]:
statsData.info()
statsData.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 42 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   team                  240 non-null    object 
 1   wins                  240 non-null    float64
 2   losses                240 non-null    float64
 3   goals                 240 non-null    float64
 4   total_yel_card        240 non-null    float64
 5   total_red_card        240 non-null    float64
 6   total_scoring_att     240 non-null    float64
 7   ontarget_scoring_att  240 non-null    float64
 8   hit_woodwork          240 non-null    float64
 9   att_hd_goal           240 non-null    float64
 10  att_pen_goal          240 non-null    float64
 11  att_freekick_goal     240 non-null    float64
 12  att_ibox_goal         240 non-null    float64
 13  att_obox_goal         240 non-null    float64
 14  goal_fastbreak        240 non-null    float64
 15  total_offside         2

(240, 42)

## Check for any missing values in the stats Dataframe

In [13]:
statsDataFrame = pd.DataFrame(statsData)

# Check for missing values in each column
missing_values = statsDataFrame.isnull().sum()

# Print the number of missing values for each column
print(missing_values)
print()

# Filter out columns that have missing values and print them
missing_values_filtered = missing_values[missing_values > 0]
if len(missing_values_filtered) > 0:
    print("Columns with missing values and their count:")
    print(missing_values_filtered)
else:
    print("There are no missing values in any column.")

team                     0
wins                     0
losses                   0
goals                    0
total_yel_card           0
total_red_card           0
total_scoring_att        0
ontarget_scoring_att     0
hit_woodwork             0
att_hd_goal              0
att_pen_goal             0
att_freekick_goal        0
att_ibox_goal            0
att_obox_goal            0
goal_fastbreak           0
total_offside            0
clean_sheet              0
goals_conceded           0
saves                   20
outfielder_block         0
interception             0
total_tackle             0
last_man_tackle          0
total_clearance          0
head_clearance          20
own_goals                0
penalty_conceded         0
pen_goals_conceded       0
total_pass               0
total_through_ball      20
total_long_balls         0
backward_pass           80
total_cross              0
corner_taken             0
touches                  0
big_chance_missed       80
clearance_off_line       0
d

## Check the skewness for each of the columns with missing values

### Total of 6 columns with missing values will be placed in a pd DataFrame

In [16]:
missingValueColumns = pd.DataFrame(statsDataFrame[['saves', 'head_clearance', 'total_through_ball', 'backward_pass', 'big_chance_missed', 'dispossessed']])
missingValueColumns.skew()

saves                 0.392526
head_clearance        0.519139
total_through_ball    1.725295
backward_pass         0.713525
big_chance_missed     1.065818
dispossessed          0.639976
dtype: float64

### Analysis:

saves (0.392526): This shows a moderate positive skew, suggesting that most of the data are concentrated on the lower end, with fewer high values extending the tail to the right.

head_clearance (0.519139): Also indicates a moderate positive skew. Similar to "saves", most data points are lower, with some high values stretching the distribution to the right.

total_through_ball (1.725295): This has a high positive skewness, indicating a significant number of lower values and a long tail towards the higher values. This suggests that very high values are relatively rare but significantly impact the distribution's shape.

backward_pass (0.713525): Shows a positive skew but less extreme than total_through_ball. It indicates a concentration of data towards lower values with a tail of higher values.

big_chance_missed (1.065818): With a positive skewness greater than 1, this distribution has a long right tail. There are significantly more lower values, with the higher values stretching the distribution.

dispossessed (0.639976): Exhibits a moderate positive skew, indicating a concentration of lower values with a tail of higher values, but not as pronounced as total_through_ball or big_chance_missed.

## Adding of new column "Draw" to the statsDataFrame

### Upon reviewing the statsDataFrame, we noticed the absence of a "draw" column that would reflect the number of draws a football team had during the season. Considering each team plays 38 matches annually, we computed the "draw" figures by deducting the sum of wins and losses from 38. After calculating this data, we would integrate this new "draw" column into the existing dataFrame.

In [25]:
# Calculate the "draw" column first
statsDataFrame['draw'] = 38 - statsDataFrame['wins'] - statsDataFrame['losses']

# Get the position of the 'losses' column
# The new column will be inserted at position + 1
position = statsDataFrame.columns.get_loc('losses') + 1

# Insert the 'draw' column next to 'losses'
statsDataFrame.insert(position, 'draw_temp', statsDataFrame['draw'])

# Drop the original 'draw' column and rename 'draw_temp' to 'draw'
statsDataFrame.drop('draw', axis=1, inplace=True)
statsDataFrame.rename(columns={'draw_temp': 'draw'}, inplace=True)

statsDataFrame.shape

(240, 43)

### statsDataFrame now has 43 columns and 240 rows